File: app/src/main/java/com/kalidratorma/yssapp/adapter/PlayerAdapter.java
package com.kalidratorma.yssapp.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.kalidratorma.yssapp.R;
import com.kalidratorma.yssapp.model.Player;

import java.util.List;

public class PlayerAdapter extends RecyclerView.Adapter<PlayerAdapter.PlayerViewHolder> {
    private List<Player> playerList;

    public PlayerAdapter(List<Player> playerList) {
        this.playerList = playerList;
    }

    @NonNull
    @Override
    public PlayerViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
        View view = inflater.inflate(R.layout.player_item, parent, false);
        return new PlayerViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull PlayerViewHolder holder, int position) {
        holder.playerItemTextView.setText(playerList.get(position).getName());
    }

    @Override
    public int getItemCount() {
        return playerList.size();
    }

    class PlayerViewHolder extends RecyclerView.ViewHolder {

        TextView playerItemTextView;

        public PlayerViewHolder(@NonNull View itemView) {
            super(itemView);

            playerItemTextView = itemView.findViewById(R.id.playerItemTextView);
        }
    }
}

File: app/src/main/java/com/kalidratorma/yssapp/adapter/PlayerAdapter.kt
package com.kalidratorma.yssapp.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.kalidratorma.yssapp.MainActivity
import com.kalidratorma.yssapp.R
import com.kalidratorma.yssapp.adapter.PlayerAdapter.PlayerViewHolder
import com.kalidratorma.yssapp.model.Player

class PlayerAdapter(private val playerList: List<Player>) : RecyclerView.Adapter<PlayerViewHolder>() {
    var onItemClick: ((Player) -> Unit)? = null

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): PlayerViewHolder {
        val inflater = LayoutInflater.from(parent.context)
        val view = inflater.inflate(R.layout.player_item, parent, false)
        return PlayerViewHolder(view)
    }

    override fun onBindViewHolder(holder: PlayerViewHolder, position: Int) {
        holder.playerItemTextView.text = playerList[position].name
    }

    override fun getItemCount(): Int {
        return playerList.size
    }

    inner class PlayerViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var playerItemTextView: TextView

        init {
            playerItemTextView = itemView.findViewById(R.id.playerItemTextView)

            itemView.setOnClickListener {
                onItemClick?.invoke(playerList[adapterPosition])
            }
        }
    }
}